### DISCUSSION TITLE
Add flash to pyboard

### DISCUSSION THREAD

**Discussion by IHOXOHI** [micropython/micropython#14310] [STM32 / Pyboard]
Hi,
Is it possible to add flash memory to a pyboard?

Thanks.

**Comment by peterhinch** [comment]
[This repo](https://github.com/peterhinch/micropython_eeprom) enables Flash (and other chip-level memory devices) to be added to MicroPython targets.

**Reply by IHOXOHI** [reply]
Wouah!
THANKS.

**Comment by IHOXOHI** [comment]
Hey Peter,
I tried the adafruit FRAM flash model; congratulation, it works fine. It's fantastic.
I'm very interested about the use of this kind of flash memory for speed up or/and enlarge the RAM part of the processor...
Is it possible? 
Thanks a lot.

**Reply by peterhinch** [reply]
In general this is not possible. RAM has a special attribute in that machine code may be executed from there. Many chips also allow machine code to run from on-chip Flash (XIP - eXecute In Place), and some allow SPIRAM to extend RAM. These capabilities require hardware support in the processor chip.

MicroPython drivers can't do this. They can provide data storage, either as `bytearray` objects or as mountable filesystems. The latter can contain Python bytecode which can be executed in place. If files containing machine code were stored, they would have to be copied to RAM for execution.

It's also worth noting that on-chip RAM is very fast via a parallel interface. Access times for SPIRAM are likely to be slower.

**Reply by IHOXOHI** [reply]
OK.
I tried a 'computer' A20 from olimex which can use a emmc flash to store the system... It's clear that the speed increase... It looks more stable too. Strange too, because you have to keep the sd card in place...? I don't understand what's happen in this case... A mix?
THANKS.

**Reply by IHOXOHI** [reply]
The pybnano too  have a better speed, NO? Look at his processor speed! What's happen?
Today I bought the teensy4.1... 2 places to receive a 'QSPI FLASH'...?
Thanks.

**Reply by robert-hh** [reply]
The Teensy port does not yet support external flash or RAM. It's somewhere on my todo list.

**Reply by IHOXOHI** [reply]
Thanks Robert. If I can do too something... More than a pleasure...
The pimo with 8/16M of flash has a quiet different speed and stability compared to the pico which have the same processor...?
Ingeniors from pimo said that this is the same board with just 8 or 16 of flash and a usbC???
So, how an additionalf flash memory can speed up alls process of the processor?  It seems that it's possible with micropython too.
